internal/controllers/topology/cluster/desired_state.go

```diff
@@ -404,21 +404,25 @@ func (r *Reconciler) computeControlPlaneVersion(ctx context.Context, s *scope.Sc
 // while some other fields should be managed as part of the actual Cluster (e.g. Cluster.Spec.ControlPlaneRef); in this func
 // we are concerned only about the latest group of fields.
 func computeCluster(_ context.Context, s *scope.Scope, infrastructureCluster, controlPlane *unstructured.Unstructured) *clusterv1.Cluster {
-	cluster := s.Current.Cluster.DeepCopy()
-
-	// Enforce the topology labels.
-	// NOTE: The cluster label is added at creation time so this object could be read by the ClusterTopology
-	// controller immediately after creation, even before other controllers are going to add the label (if missing).
-	if cluster.Labels == nil {
-		cluster.Labels = map[string]string{}
+	cluster := &clusterv1.Cluster{
```

Comment:
Computing a trimmed down desired state won't alleviate us from the need to get allowedPaths in order, due to json serialization weirdness, so I suggest moving this change to a separated PR/out of the critical path.
Let's keep this PR a small as possible

Reply:
I reverted this changes on the PR 👍